src/app/shared/models.ts

```diff
@@ -317,12 +323,17 @@ export interface KsonnetAppSpec {
 
 export type AppSourceType = 'Helm' | 'Kustomize' | 'Ksonnet' | 'Directory';
 
+export interface DirectoryAppSpec {
```

Comment:
I guess there is one more PR for api server coming? The DirectoryAppSpec should not have `recurse` field since this is part of application spec, not an additional app information stored in git.

Reply:
Should be pushed now.